projects/bootloader/src/dispatcher.c

```diff
@@ -63,3 +63,16 @@ StatusCode dispatcher_register_callback(BootloaderDatagramId id, DispatcherCallb
 void tx_cmpl_cb(void) {
   can_datagram_start_listener(&s_datagram_rx);
 }
+
+StatusCode status_code_response(StatusCode status) {
+  CanDatagramTxConfig s_response_config = {
+    .dgram_type = BOOTLOADER_DATAGRAM_STATUS_RESPONSE,
+    .destination_nodes_len = 0,  // client listens to all datagrams
+    .destination_nodes = NULL,
+    .data_len = 1,
+    .data = &status,
```

Comment:
Careful! The `status` parameter's memory will get deallocated when the function returns, so this pointer won't be valid when can_datagram goes to access it. Either use static storage (in the function or file) or pass in a pointer and document that the pointer has to remain valid until the datagram finishes TXing.